WinImageBuilder.psm1

```diff
@@ -39,6 +41,78 @@ $VirtIODriverMappings = @{
+    $availableConfigs += Get-ConfigFromTemplate -Name "virtual_disk_format" -DefaultValue "VHDX"
+    $availableConfigs += Get-ConfigFromTemplate -Name "image_type" -DefaultValue "MAAS"
+    $availableConfigs += Get-ConfigFromTemplate -Name "disk_layout" -DefaultValue "BIOS"
+    $availableConfigs += Get-ConfigFromTemplate -Name "product_key" 
+    $availableConfigs += Get-ConfigFromTemplate -Name "extra_features" 
+    $availableConfigs += Get-ConfigFromTemplate -Name "force" -DefaultValue $false
+    $availableConfigs += Get-ConfigFromTemplate -Name "install_maas_hooks" -DefaultValue $false
+    $availableConfigs += Get-ConfigFromTemplate -Name "zip_password"
+    $availableConfigs += Get-ConfigFromTemplate -Name "administrator_password" -DefaultValue "Pa`$`$w0rd" `
+                                     -GroupName "vm"
+    $availableConfigs += Get-ConfigFromTemplate -Name "external_switch" -GroupName "vm"
+    $availableConfigs += Get-ConfigFromTemplate -Name "cpu_count" -DefaultValue "1" `
+                                     -GroupName "vm"
+    $availableConfigs += Get-ConfigFromTemplate -Name "ram_size" -DefaultValue "2048" `
+                                     -GroupName "vm"
+    $availableConfigs += Get-ConfigFromTemplate -Name "disk_size" -DefaultValue "40G" `
+                                     -GroupName "vm"
+    $availableConfigs += Get-ConfigFromTemplate -Name "virtio_iso_path" -GroupName "drivers"
+    $availableConfigs += Get-ConfigFromTemplate -Name "virtio_base_path" -GroupName "drivers"
+    $availableConfigs += Get-ConfigFromTemplate -Name "drivers_path" -GroupName "drivers"
+    $availableConfigs += Get-ConfigFromTemplate -Name "install_updates" -DefaultValue $false `
+                                     -GroupName "updates"
+    $availableConfigs += Get-ConfigFromTemplate -Name "purge_updates" -DefaultValue $false `
+                                     -GroupName "updates"
+    $availableConfigs += Get-ConfigFromTemplate -Name "run_sysprep" -DefaultValue $true `
+                                     -GroupName "sysprep"
+    $availableConfigs += Get-ConfigFromTemplate -Name "unattend_xml_path" -DefaultValue "UnattendTemplate.xml" `
+                                     -GroupName "sysprep"
+    $availableConfigs += Get-ConfigFromTemplate -Name "disable_swap" -DefaultValue $false `
+                                     -GroupName "sysprep"
+    $availableConfigs += Get-ConfigFromTemplate -Name "persist_drivers_install" -DefaultValue $true `
+                                     -GroupName "sysprep"
+
+    return $availableConfigs
+}
+
+function Get-ConfigFromTemplate {
+    param(
+        [parameter(Mandatory=$true)]
+        [string]$Name,
+        [parameter(Mandatory=$false)]
+        [string]$DefaultValue="",
+        [parameter(Mandatory=$false)]
+        [string]$GroupName="DEFAULT"
+        )
+
+    return @{'Name' = $Name;'GroupName' = $GroupName; 'DefaultValue'  = $DefaultValue}

```

Comment:
time to add IsBoolean = $false flag and use it at line 105, by adding a new parameter switch for the Get-InifileValue